src/libraries/System.Net.Http/src/System/Net/Http/Headers/ObjectCollection.cs

```diff
@@ -94,7 +94,7 @@ public void Clear()
         }
 
         public bool Contains(T item) =>
-            ReferenceEquals(item, _items) ||
+            (_size == 1 && _items!.Equals(item)) ||
```

Comment:
Note that _size can be 1 even if it's an array. The logic here will still work since Equals will return false. But I'm not sure that doing the check for _size == 1 is useful.